tests/test_metadata.py

```diff
@@ -76,6 +88,81 @@ def test_preauthencoding(self):
         # Checking for Pre-Auth-Encoding generated is correct.
         self.assertEqual(self.pae, envelope_obj.pae)
 
+    def test_sign(self):
+        """Test payload signing of envelope"""
+
+        # Create an Evnvelope with no signatures
+        envelope_dict = copy.deepcopy(self.envelope_dict)
+        envelope_dict["signatures"] = []
+        envelope_obj = Envelope.from_dict(envelope_dict)
+
+        for (_, key) in self.key_dicts:
+            # Test with invalid signer
+            with self.assertRaises(TypeError):
+                envelope_obj.sign(key)
```

Comment:
I am a little confused.
Why this signer is invalid? Because it's actually a `key` instance or?

Reply:
Sorry for your confusion here, it was `key_dict` and I changed the variable name now.